benchmarks/bench_sdbaseline.py

```diff
@@ -0,0 +1,731 @@
+from casatasks.private.task_sdbaseline import check_fftthresh, is_empty, parse_wavenumber_param
+from casatestutils import selection_syntax
+
+
+tb = table()
+ctsys_resolve = ctsys.resolve
+
+# ASV iteration control (https://asv.readthedocs.io/en/stable/benchmarks.html#benchmark-attributes)
+number = 1            # i.e., always run the setup and teardown methods
+repeat = (1, 2, 30.0) # between 1 and 2 iterations per round w/ soft cutoff (start no new repeats) past 1m
+rounds = 3            # amount of instances a "repeat block" is run to collect samples
+min_run_count = 5     # enforce the min_repeat * rounds setting is met
+timeout = 3600        # conservative 1hr hard cap for duration of a single test execution
+
+### Utilities for reading blparam file
+class FileReader(object):
+    def __init__(self, filename):
+        self.__filename = filename
+        self.__data = None
+        self.__nline = None
+
+    def read(self):
+        if self.__data is None:
+            f = open(self.__filename, 'r')
+            self.__data = f.readlines()
+            f.close()
+            self.__nline = len(self.__data)
+        return
+
+    def nline(self):
+        self.read()
+        return self.__nline
+
+    def index(self, txt, start):
+        return self.__data[start:].index(txt) + 1 + start
+
+    def getline(self, idx):
+        return self.__data[idx]
+
+
+class BlparamFileParser(FileReader):
+    def __init__(self, blfile):
+        FileReader.__init__(self, blfile)
+        #self.__nrow = None
+        #self.__coeff = None
+        #self.__rms = None
+        self.__ctxt = 'Baseline parameters\n'
+        self.__rtxt = 'Results of baseline fit\n'
+
+def remove_single_file_dir(filename):
+    """
+    Remove a single file or a single directory.
+    For filename, '.' and those end with '..' (namely, '..', '../..' etc.)
+    are not allowed.
+    """
+    if filename == '.' or filename[-2:] == '..':
+        raise Exception("Caution! Attempting to remove '" + filename + "'!!")
+    
+    if os.path.exists(filename):
+        if os.path.isdir(filename):
+            shutil.rmtree(filename)
+        else: # file or symlink
+            os.remove(filename)
+
+
+def remove_files_dirs(filename):
+    """
+    Remove files/directories/symlinks 'filename*'.
+    For filename, '', '.' and those end with '..' (namely, '..', '../..' etc.)
+    are not allowed.
+    """
+    if filename == '.' or filename[-2:] == '..':
+        raise Exception("Caution! Attempting to remove '" + filename + "*'!!")
+    elif filename == '':
+        raise Exception("The parameter 'filename' must not be a null string.")
+    
+    filenames = glob.glob('{}*'.format(filename.rstrip('/')))
+
+    for filename in filenames:
+        remove_single_file_dir(filename)
+
+def _get_num_files_by_keyword(cmpresult, keyword):
+    return len(sum([eval(s[s.index(':')+1:]) for s in cmpresult if s.startswith(keyword)], []))
+
+
+class sdbaseline_unittest_base(unittest.TestCase):
```

Comment:
Rename this class to
class BaseSetup() or DataSetUp something similar. You can also remove the unittest.TestCase inheritance.

Reply:
Thanks for taking a look at this Sandra! I've put in all the changes that are mentioned in the comments, and I was able to remove both the classes that you mentioned.